inc/sitemaps/class-sitemaps-admin.php

```diff
@@ -59,8 +59,12 @@ public function status_transition( $new_status, $old_status, $post ) {
 			return;
 		}
 
-		if ( WP_CACHE ) {
-			wp_schedule_single_event( ( time() + 300 ), 'wpseo_hit_sitemap_index' );
+		if ( WP_CACHE && ! wp_next_scheduled( 'wpseo_hit_sitemap_index' ) ) {
```

Comment:
Why not invert this `if` and return earlier?

Reply:
It's already reverted. I haven't use an early return because there is some logic to be executed as well.